rancher/rancher
# Pin the rancher-webhook chart to an exact version

## Issue: https://github.com/rancher/rancher/issues/40159
 
## Problem
Rancher specifies the minimum version of the webhook in its configuration. When there is a newer version of the chart, Rancher will try to install it, which we don't always want, since there might be incompatibilities of Rancher and a given version of the webhook.
 
## Solution
Install only the webhook specified in configuration. Do not use the notion of a "minimum" version for the webhook chart. Instead, use exact version that doesn't change automatically for a given version of Rancher - it's to remain fixed.
 
## Testing


## Engineering Testing
### Manual Testing
1. Run Rancher with the webhook chart version specified with `CATTLE_RANCHER_WEBHOOK_VERSION`. Ensure that Rancher installs that version only, whether there are any newer versions or not.
1. In the UI, upgrade or downgrade to another version. Ensure that after some time (depending on the configurable chart refresh interval), Rancher switches back to the original version.
1. Rerun Rancher with a different version specified. Ensure Rancher installs that version of the webhook chart.

### Automated Testing


## QA Testing Considerations

 
### Regressions Considerations
Ensure no changes occur for other system charts, like fleet. For them, the minimum version should behave as before, that is, Rancher may upgrade to a newer version if one is available.
